pinot-common/src/main/java/org/apache/pinot/common/utils/TarGzCompressionUtils.java

```diff
@@ -163,8 +182,12 @@ public static List<File> untar(InputStream inputStream, File outputDir)
           if (!parentFile.isDirectory() && !parentFile.mkdirs()) {
             throw new IOException(String.format("Failed to create directory: %s", parentFile));
           }
-          try (OutputStream out = Files.newOutputStream(outputFile.toPath())) {
-            IOUtils.copy(tarGzIn, out);
+          try (FileOutputStream out = new FileOutputStream(outputFile.toPath().toString())) {
```

Comment:
Curious why this change ?

Reply:
Because we need to get the FileDescriptor from the `FileOutputStream` in `streamCopyWithRateLimiter`. Creating a FileOutputStream from `outputFile` should be safe here.